src/android/CodePushPreferences.java

```diff
@@ -67,15 +67,31 @@ public InstallOptions getPendingInstall() {
 
         SharedPreferences preferences = context.getSharedPreferences(CodePushPreferences.PENDING_INSTALL_PREFERENCE, Context.MODE_PRIVATE);
         int installMode = preferences.getInt(CodePushPreferences.INSTALL_MODE_KEY, -1);
-        int rollbackTimeout = preferences.getInt(CodePushPreferences.ROLLBACK_TIMEOUT_KEY, -1);
 
-        if ((installMode != -1) && (rollbackTimeout != -1)) {
-            pendingInstall = new InstallOptions(rollbackTimeout, InstallMode.fromValue(installMode));
+        if (installMode != -1) {
+            pendingInstall = new InstallOptions(InstallMode.fromValue(installMode));
         }
 
         return pendingInstall;
     }
 
+    public void saveNotConfirmedInstall() {
+        SharedPreferences preferences = context.getSharedPreferences(CodePushPreferences.INSTALL_NOT_CONFIRMED_PREFERENCE, Context.MODE_PRIVATE);
+        SharedPreferences.Editor editor = preferences.edit();
+        editor.putBoolean(CodePushPreferences.INSTALL_NOT_CONFIRMED_KEY, true);
+        editor.commit();
+    }
+
+    public void clearNotConfirmedInstall() {
+        this.clearPreferences(CodePushPreferences.INSTALL_NOT_CONFIRMED_PREFERENCE);
+    }
+
+    public boolean isNotConfirmedInstall() {
+        SharedPreferences preferences = context.getSharedPreferences(CodePushPreferences.INSTALL_NOT_CONFIRMED_PREFERENCE, Context.MODE_PRIVATE);
+        boolean notConfirmedInstall = preferences.getBoolean(CodePushPreferences.INSTALL_NOT_CONFIRMED_KEY, false);
+        return notConfirmedInstall;
+    }
+
```

Comment:
I don't know if its too much trouble, but can we instead of saving notConfirmed save values like isConfirmed and flip the values. I think it would be easy to follow that way. If its a lot of change and could potentially delay our deliveries, I would leave it.